install.sh

```diff
@@ -277,8 +277,7 @@ EOABORT
 )"
   elif version_lt "$macos_version" "10.9"; then
     abort "Your OS X version is too old"
-  elif ! [[ "$(dsmemberutil checkmembership -U "$USER" -G "$GROUP")" = *"user is a member"* ]]; then
-    abort "This script requires the user $USER to be an Administrator."
+  elif have_sudo_access; then :
```

Comment:
Feels a bit weird being in here. What about calling `have_sudo_access` at line 305 (just above "this script will install")

Reply:
Keeping it at 305 will make it run on linux machines too, line 241 is already checking sudo for linux machines. 
I think for macOS we need to check between line 271 and 304 which is within the if statement.
Am i missing something ?

Reply:
Typo: not line 241, line 249 i meant

Reply:
Yeh, you're right, good catch. I think it would be good to unify the `sudo` checking as much as possible. Not sure why `sudo_output` and `sudo_exit_code` should be checked at all rather than just using `have_sudo_access` there? Would be nice to have one check location that's shared between macOS and Linux. Thoughts?

Reply:
Right, i think there is bug there as well, in case when SUDO_ASKPASS is set, the sudo checks would fail on line 241. 
I think we should definitely refactor the sudo checking, can we submit the refactoring as a separate PR? For this PR i am thinking about fixing the standard user run issue at line 280?
What do you think?

Reply:
I reckon roll it all into this PR (if you're up for that)!

Reply:
Cool. Will refactor that as well

Reply:
Refactoring done. Actions pre and post sudo check are different for macOS and Linux, tried to bring them as close as they get without changing the behaviour of the script.
Tested this with linux and macOS, sudo(NOPASSWD and without NOPASSWD), non sudo users and users who are not admins but still have sudo rights. Works fine in all cases now.

Reply:
Tested it, works on my local.
Although whoever is running the script as non admin user(with sudo) needs to know that all directories within /usr/local will have their non admin user as owner of these directories. I did not change this as the script prompts the user before making this change.

If this looks fine to you, its ready to be merged.

Reply:
Fine with me, great work 👏!

Reply:
Thanks guys, sorry I haven't had any time to test, but the changes look good :).